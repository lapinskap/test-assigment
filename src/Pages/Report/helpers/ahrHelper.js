export const getAhrUrl = (url, isAhr) => {
  let result = isAhr ? '/ahr' : '';
  result += url;

  return result;
};

export const asd = () => {};
