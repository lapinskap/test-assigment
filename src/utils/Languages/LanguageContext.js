import React, {
  useContext, useEffect, useState,
} from 'react';
import { defaultLanguage } from '../Translations/translationUtils';

const LanguageContext = React.createContext({
  languagesConfig: [],
  defaultLanguage: '',
});

export const LANGUAGE_STORAGE_KEY = 'user_language';

export const getLanguage = () => {
  const value = window.localStorage.getItem(LANGUAGE_STORAGE_KEY);
  if (value) {
    return value;
  }
  return defaultLanguage;
};
export const setLanguage = (lang) => {
  window.localStorage.setItem(LANGUAGE_STORAGE_KEY, lang);
};
export const removeLanguage = () => {
  window.localStorage.removeItem(LANGUAGE_STORAGE_KEY);
};

export const useLanguages = (withDefault = false, asOptions = false) => {
  const [result, setResult] = useState([]);
  const { languagesConfig } = useContext(LanguageContext);
  useEffect(() => {
    if (withDefault) {
      setResult(languagesConfig);
    } else {
      setResult(languagesConfig.filter(({ isDefault }) => !isDefault));
    }
  }, [withDefault, languagesConfig]);
  if (!asOptions) {
    return result;
  }
  return result.map(({ code, label }) => ({ value: code, label }));
};

export const useDefaultLanguage = () => {
  const { defaultLanguage: result } = useContext(LanguageContext);

  return result;
};

export default LanguageContext;
