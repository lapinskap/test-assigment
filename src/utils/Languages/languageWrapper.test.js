import React from 'react';
import { create } from 'react-test-renderer';
import LanguageWrapper, { loadInterfaceTranslation } from './languageWrapper';
import { wait } from '../../tests/helpers';
import '../../tests/setupTests';
import { LANGUAGE_STORAGE_KEY } from './LanguageContext';

describe('languageWrapper', () => {
  afterEach(() => {
    window.translations = null;
    window.localStorage.removeItem(LANGUAGE_STORAGE_KEY);
  });

  it('fetches interfeace translations', async () => {
    await loadInterfaceTranslation('en');
    expect(Object.keys(window.translations).length).toBe(0);
  });

  test('LanguageWrapper does nothing for default language', async () => {
    window.localStorage.removeItem(LANGUAGE_STORAGE_KEY);
    const wrapper = create((
      <LanguageWrapper><input /></LanguageWrapper>
    ));
    await wait(100);
    wrapper.update();
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('LanguageWrapper render correctly for no default language', async () => {
    window.localStorage.setItem(LANGUAGE_STORAGE_KEY, 'en');
    const wrapper = create((
      <LanguageWrapper><input /></LanguageWrapper>
    ));
    await wait(100);
    wrapper.update();
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
