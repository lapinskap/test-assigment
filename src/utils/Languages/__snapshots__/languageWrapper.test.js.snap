// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languageWrapper LanguageWrapper does nothing for default language 1`] = `null`;

exports[`languageWrapper LanguageWrapper render correctly for no default language 1`] = `null`;
